DifferentialDrive.md

```diff
@@ -0,0 +1,545 @@
+# DIFFERENTIAL DRIVE
+
+![alt text](ezgif.com-video-to-gif-converter.gif)
+This project is going to cover how to make both a basic and an advanced differential drive. 
+
+Prereqs: A good understanding of java, command-based programming, and how to use git in vscode. 
+
+Goals: Get an understanding of how to create a differential drive with advance controls. 
+
+---
+
+## Initial Setup
+
+### Robot Code Structure
+
+Robot code is structured using **subsystems** and **commands** within the **Command-Based Framework**. This architecture simplifies robot programming by breaking down complex tasks into smaller, manageable pieces:
+
+- **Subsystems**: These represent individual components or systems on the robot, like the drivetrain or arm mechanism. They encapsulate the hardware and related functionalities. [Learn more about subsystems here](https://docs.wpilib.org/en/stable/docs/software/commandbased/subsystems.html).
+
+- **Commands**: Commands are actions that the robot can perform, such as driving forward or rotating. These actions often involve multiple subsystems working together. [Learn more about commands here](https://docs.wpilib.org/en/stable/docs/software/commandbased/commands.html).
+
+For a deeper dive into command-based programming, check out [this guide](https://docs.wpilib.org/en/stable/docs/software/commandbased/what-is-command-based.html).
+
+### Setting up VScode
+
+After making a new branch for this project and opening up wpilib vscode, start by making a drive folder. Which then will have a drive.java file and a driveConstants.java file. 
```

Comment:
That's actually something I was wondering about -- where do we want to have the setting up your computer stuff? We might want to put that in a separate doc.

And then what should we do for the intro to java docs? Should I have them use wpilib vscode for that somehow? I was thinking they'd install regular vscode for that, since it's good to have anyway, but they won't be using it for any of the other sciguides projects so idk